packages/cli/src/commands/update.ts

```diff
@@ -0,0 +1,122 @@
+import fs from "fs";
+import path from "path";
+import inquirer from "inquirer";
+import { run, RunOptions } from "npm-check-updates";
+import { installDependencies } from "../utils/installDependencies.js";
+import { Index } from "npm-check-updates/build/src/types/IndexType.js";
+
+function getPackageJSON(projectPath: string) {
+  const packageJsonPath = path.join(projectPath, "package.json");
+  if (!fs.existsSync(packageJsonPath)) {
+    console.error(`package.json not found in the ${projectPath} directory.`);
+    return;
+  }
+  return JSON.parse(fs.readFileSync(packageJsonPath, "utf8"));
+}
+
+function setPackageJSON(projectPath: string, updatedPackageJSON: Index) {
+  const packageJsonPath = path.join(projectPath, "package.json");
+  if (!fs.existsSync(packageJsonPath)) {
+    console.error(`package.json not found in the ${projectPath} directory.`);
+    return;
+  }
+  fs.writeFileSync(packageJsonPath, JSON.stringify(updatedPackageJSON, null, 2), "utf8");
+  return;
+}
+
+export async function updateCommand(projectPath: string) {
+  const triggerDevPackage = "@trigger.dev";
+  const packageData = getPackageJSON(projectPath);
+
+  if (!packageData) {
+    return;
+  }
+
+  const packageMaps: { [k: string]: { type: string; version: string } } = {};
+  const packageDependencies = packageData.dependencies || {};
+  const packageDevDependencies = packageData.devDependencies || {};
+  Object.keys(packageDependencies).forEach((i) => {
+    packageMaps[i] = { type: "dependencies", version: packageDependencies[i] };
+  });
+  Object.keys(packageDevDependencies).forEach((i) => {
+    packageMaps[i] = {
+      type: "devDependencies",
+      version: packageDevDependencies[i],
+    };
+  });
+
+  // Use npm-check-updates to get updated dependency versions
+  const ncuOptions: RunOptions = {
+    packageData,
+    upgrade: true,
+    jsonUpgraded: true,
+  };
+
+  // Can either give a json like package.json or just with deps and their new versions
+  const updatedDependencies: Index | void = await new Promise((resolve, reject) =>
```

Comment:
There is no need to explicitly type this or create a promise like you are doing. You can just do:

```ts
const updatedDependencies = await run(ncuOptions)
```

Reply:
What you're suggesting won't be doable as run is not exported as an async function from the library itself.

Reply:
Are you sure? According to their docs it is

Reply:
Yes, I tried that. Wait let me share the function declaration.

Reply:
<img width="661" alt="Screenshot 2023-08-22 at 2 10 11 AM" src="https://github.com/triggerdotdev/trigger.dev/assets/46300090/70b7a72e-ca09-4f1b-b4ae-6038e18a995a">

Reply:
It's a function that returns a promise, but isn't async. Hence, need to await the promise resolution.

Reply:
Any function that returns a promise can be awaited. You only use the `async` keyword when defining a function if you want to **use** await inside the function:

```ts
function foo(): Promise<string> {
  return new Promise((resolve, reject) => resolve("foo"));
}

async function bar() {
  const baz = await foo();

  console.log(baz); // logs "foo"
}
```

Reply:
Oh, I see. Thanks for teaching me this. I just relied on the function declaration always.

Reply:
I've pushed as taught.